File: week2/4_least_common_multiple/lcm.py
# Uses python3
import sys

def lcm_naive(a, b):
    for l in range(1, a*b + 1):
        if l % a == 0 and l % b == 0:
            return l

    return a*b

def gcd_r(a,b):
    if a==0:
        return b
    elif b==0:
        return a
    else:
        return gcd_r(b, a%b)

def lcm_fast(a,b):
    """
    Based on this formula
    a*b = gcd*lcm
    """
    return int(a/gcd_r(a,b)) * int(b)

if __name__ == '__main__':
    #input = sys.stdin.read()
    a, b = map(int, input().split())
    #print(lcm_naive(a, b))
    print(lcm_fast(a, b))

File: week2/LCM.java
import java.util.*;

public class LCM {

  private static long lcm_naive(int a, int b) {
    for (long l = 1; l <= (long) a * b; ++l)
      if (l % a == 0 && l % b == 0)
        return l;

    return (long) a * b;
  }

  private static long gcd_r(long a, long b) {
    if (a == 0){
        return b;
    } else if (b == 0){
        return a;
    } else {
        return gcd_r(b, a%b);
    }
  }

  private static long lcm_fast(long a, long b) {
    return (a*b) / gcd_r(a, b);
  }

  public static void main(String args[]) {
    Scanner scanner = new Scanner(System.in);
    int a = scanner.nextInt();
    int b = scanner.nextInt();

    System.out.println(lcm_fast(a, b));
    scanner.close();
  }
}
